### ISSUE TITLE
False Positive UnnecessaryApply

### ISSUE THREAD

**Issue by celikkoseoglu** [arturbosch/detekt#1643]
Following code gets flagged.

```kotlin
private fun getDecimalFormatter(): DecimalFormat {
        val nf = NumberFormat.getNumberInstance(applicationInfo.localeObj)
        return if (nf is DecimalFormat) {
            @Suppress("UnnecessaryApply")
            nf.apply { applyPattern(DECIMAL_FORMAT_PATTERN) }
        } else {
            DecimalFormat(DECIMAL_FORMAT_PATTERN)
        }
}
```

The way to write this without the apply block is

```kotlin
private fun getDecimalFormatter(): DecimalFormat {
        val nf = NumberFormat.getNumberInstance(applicationInfo.localeObj)
        return if (nf is DecimalFormat) {
            nfapplyPattern(DECIMAL_FORMAT_PATTERN)
            nf
        } else {
            DecimalFormat(DECIMAL_FORMAT_PATTERN)
        }
}
```

The following works as well since we don't have UnnecessaryAlso

```kotlin
private fun getDecimalFormatter(): DecimalFormat {
        val nf = NumberFormat.getNumberInstance(applicationInfo.localeObj)
        return if (nf is DecimalFormat) {
            // If we create DecimalFormat on our own, it takes default US locale (ie currency)
            // Here we use localised formatter to support currency of the device locale
            nf.also { it.applyPattern(DECIMAL_FORMAT_PATTERN) }
        } else {
            DecimalFormat(DECIMAL_FORMAT_PATTERN)
        }
    }
```

**Comment by 3flex** [comment]
Can you take a quick look at #1539 and see if that's the same issue?

**Comment by celikkoseoglu** [comment]
Yes, it is the same issue. Closing this one.
